python/essex-config/essex_config/config.py

```diff
@@ -131,10 +140,20 @@ def _load_config(
             if prefix_annotation is None:
                 field_prefix += f".{name}" if field_prefix != "" else name
             values[name] = _load_config(
-                field_type, sources, prefix=field_prefix, inner=True
+                field_type,
+                sources,
+                prefix=field_prefix,
+                inner=True,
+                parse_env_values=parse_env_values,
             )
             continue
 
+        env_values: dict[str, Any] = {}
+        for source in sources:
+            if isinstance(source, EnvSource):
+                extra_data = source.get_data()
+                env_values = {**extra_data, **env_values}
```

Comment:
you could use a filter here, something like:
```python
for source in filter(lambda src: isinstance(src, EnvSource), sources):
    ...
```
that way you can avoid the `if` inside the loop